detox/src/Detox.js

```diff
@@ -127,6 +101,8 @@ class Detox {
   }
 
   async afterEach(testSummary) {
+    await this._assertNoPendingInit();
```

Comment:
c&p: really wish you'd cover this logic using unit tests

Reply:
Covered with test: `Calling detox.afterEach() before detox.init() completes, throws error`